app/policies/application_policy.rb

```diff
@@ -4,7 +4,7 @@ class ApplicationPolicy
   attr_reader :user, :record
 
   def initialize(user, record)
-    @user = user
+    @user = user || Null::User.new
```

Comment:
I am assuming the use case for the `Null::User.new` is for users not signed in. and pages that include authorization (but allow non signed in users). For this case I would just have a `return false if user.blank?` on these specific areas. I don't like what this object is doing at all (and it isn't entirely correct either)